platform/mellanox/mlnx-platform-api/sonic_platform/thermal.py

```diff
@@ -420,3 +437,123 @@ def get_high_critical_threshold(self):
         if self.category == THERMAL_DEV_CATEGORY_MODULE and value_float == THERMAL_API_INVALID_HIGH_THRESHOLD:
             return None
         return value_float / 1000.0
+
+
+    @classmethod
+    def _write_generic_file(cls, filename, content):
+        """
+        Write a generic file if content changed
+        """
+        try:
+            with open(filename, 'w+') as file_obj:
+                origin_content = file_obj.read()
+                if origin_content != content:
+                    file_obj.write(content)
+        except Exception as e:
+            logger.log_info("Fail to write file {} due to {}".format(filename, repr(e)))
+
+    @classmethod
+    def set_thermal_algorithm_status(cls, status, force=True):
```

Comment:
what is the behaviour expected when disabled?

Reply:
When disable the thermal algorithm, kernel no longer adjust the FAN speed according to thermal zone temperature. We usually disable the thermal algorithm when FAN absence or PSU absence. If a fan unit or PSU is removed from switch, there will be a "air hole" at switch, we need set fan speed to 100% and prevent thermal algorithm adjust it.